File: pset6/readability/readability.py
from cs50 import get_string

# A function to calculate Colemna-Liau index


def readability(text):
    sentences = 0
    words = 1
    letters = 0

    for i in range(len(text)):
        if text[i] == '!' or text[i] == '.' or text[i] == "?":
            sentences += 1
        if text[i] == ' ' and text[i+1]:
            words += 1
        if text[i].isalpha():
            letters += 1
    L = (letters * 100) / words
    S = (sentences * 100) / words

    index = (0.0588 * L) - (0.296 * S) - 15.8
    return round(index)


# Asking user to enter a piece of text
input_text = get_string("Text: ")
grade = readability(input_text)

# Checking for grade level that can understand the piece of text
if grade >= 16:
    print('Grade 16+')
elif grade < 1:
    print('Before Grade 1')
else:
    print(f'Grade {grade}')

File: pset2/readability/readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<ctype.h>
#include<math.h>

float readability(string text);

int main(void)
{
    string text = get_string("Text: ");
    float grade = readability(text);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(grade));
    }
}

float readability(string text)
{
    int sentences = 0, words = 1, letters = 0;
    int n = strlen(text);
    float L = 0, S = 0, index = 0;

    for (int i = 0; i < n ; i++)
    {
        // Counting the number of sentenses
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
        // Counting the number of words
        if (text[i] == ' ')
        {
            words++;
        }
        // Checking if the character is alphabet
        if (isalpha(text[i]))
        {
            letters++;
        }

    }
    // computing for L
    L = (letters * 100) / (float) words;
    // computing for S
    S = (sentences * 100)  / (float) words;

    // Computing for Index
    index = (0.0588 * L) - (0.296 * S) - 15.8;

    return index;

}